homeassistant/components/light/hue.py

```diff
@@ -345,3 +364,124 @@ def turn_off(self, **kwargs):
+                 bridge_type):
+        """Initialize the light group."""
+        self.lightgroup_id = lightgroup_id
+        self.info = info
+        self.bridge = bridge
+        self.update_lights = update_lights
+        self.bridge_type = bridge_type
+
+    @property
+    def unique_id(self):
+        """Return the ID of this Hue light group."""
+        return "{}.{}".format(
+            self.__class__, self.info.get('uniqueid', self.name))
+
+    @property
+    def name(self):
+        """Return the name of the Hue light group."""
+        return self.info.get('name', DEVICE_DEFAULT_NAME)
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return self.info['action'].get('bri')
+
+    @property
+    def xy_color(self):
+        """Return the XY color value."""
+        return self.info['action'].get('xy')
+
+    @property
+    def color_temp(self):
+        """Return the CT color value."""
+        return self.info['action'].get('ct')
+
+    @property
+    def is_on(self):
+        """Return true if any light of the group is on."""
+        self.update_lights()
+
+        return self.info['state']['any_on']
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_HUE
+
+    def turn_on(self, **kwargs):
+        """Turn the specified or all lights on."""
+        command = {'on': True}
+
+        if ATTR_TRANSITION in kwargs:
+            command['transitiontime'] = kwargs[ATTR_TRANSITION] * 10
+
+        if ATTR_XY_COLOR in kwargs:
+            command['xy'] = kwargs[ATTR_XY_COLOR]
+        elif ATTR_RGB_COLOR in kwargs:
+            xyb = color_util.color_RGB_to_xy(
+                *(int(val) for val in kwargs[ATTR_RGB_COLOR]))
+            command['xy'] = xyb[0], xyb[1]
+            command['bri'] = xyb[2]
+
+        if ATTR_BRIGHTNESS in kwargs:
+            command['bri'] = kwargs[ATTR_BRIGHTNESS]
+
+        if ATTR_COLOR_TEMP in kwargs:
+            command['ct'] = kwargs[ATTR_COLOR_TEMP]
+
+        flash = kwargs.get(ATTR_FLASH)
+
+        if flash == FLASH_LONG:
+            command['alert'] = 'lselect'
+            del command['on']
+        elif flash == FLASH_SHORT:
+            command['alert'] = 'select'
+            del command['on']
+        elif self.bridge_type == 'hue':
+            command['alert'] = 'none'
+
+        effect = kwargs.get(ATTR_EFFECT)
+
+        if effect == EFFECT_COLORLOOP:
+            command['effect'] = 'colorloop'
+        elif effect == EFFECT_RANDOM:
+            command['hue'] = random.randrange(0, 65535)
+            command['sat'] = random.randrange(150, 254)
+        elif self.bridge_type == 'hue':
+            command['effect'] = 'none'
+
+        self.bridge.set_group(self.lightgroup_id, command)
+
+    def turn_off(self, **kwargs):
+        """Turn the specified or all lights off."""
+        command = {'on': False}
+
+        if ATTR_TRANSITION in kwargs:
```

Comment:
It can not be fully reused, the `xy_color`, `brightness` and `color_temp` are fetched from a different dict. I also renamed `light_id to `lightgroup_id` to better match what it is.

I will change the code to reuse the `turn_on` and `turn_off`, but keep the `HueLightGroup` class for the overriden functions. I could put everything in a constructor, but kind of defeats the purpose of inheritance of classes...